R/createImpactFeatures.R

```diff
@@ -69,32 +94,39 @@ createImpactFeatures.data.frame = function(obj, target = character(0L), cols = N
 
   })
   names(value.table) = work.cols
-  #replace values in the data
-  for (wc in work.cols) {
+  
 
+  # replace values in the data
+  for (wc in work.cols) {
     tab = value.table[[wc]]
-    if (ncol(tab) == 2) {#regression or binary classif
+    if (ncol(tab) == 2) {# regression or binary classif
+      obj[, wc] = as.factor(obj[, wc])
       levels(obj[, wc]) = tab[, 2]
-      obj[, wc] = as.numeric(as.character(obj[, wc]))
-    } else { # for multiclass classif
-      new.cols = paste(wc, colnames(tab)[-1], sep = ".")
-      obj[, new.cols] = obj[, wc] #FIXME this converts all new.cols columns to character... annoying
+      n = as.numeric(table(obj[, wc])[as.character(obj[, wc])])
+      obj[, wc] = impactEncodingLambda(n, slope.param, trust.param) * as.numeric(as.character(obj[, wc])) + (1 - impactEncodingLambda(n, slope.param, trust.param)) * prior.table
+    } else { # multiclass classif
+      new.cols = paste(wc, colnames(tab[, -1]), sep = ".")
+      obj[, new.cols] = obj[, wc]
       obj[, wc] = NULL
       for(i in seq_along(new.cols)) {
-          obj[,new.cols[i]] = as.factor(obj[,new.cols[i]]) # see above
-          levels(obj[, new.cols[i]]) = tab[, i + 1]
-          obj[, new.cols[i]] = as.numeric(as.character(obj[,new.cols[i]]))
+        obj[, new.cols[i]] = as.factor(obj[, new.cols[i]])
+        levels(obj[, new.cols[i]]) = tab[order(as.character(tab[, 1])), i + 1]
+        n = as.numeric(table(obj[, new.cols[i]])[as.character(obj[, new.cols[i]])])
```

Comment:
helper function, this was used a few times already

Reply:
like that?
````
impactMulticlass = function(obj, col, tab, i) {
  obj[, col] = as.factor(obj[, col])
  levels(obj[, col]) = tab[order(as.character(tab[, 1])), i + 1]
  n = as.numeric(table(obj[, col)[as.character(obj[, col])])
  return(list(obj = obj[, col], n = n))
}
````
so the lines above would reduce to
````
obj[, new.cols[i]] = impactMulticlass(obj, new.cols[i], tab, i)$obj
n = impactMulticlass(obj, new.cols[i], tab, i)$n
````
we use this two times and would need a similar helper function for binary/regression which is technically not a problem. Do you think saving one line each is worth doing it or is more confusing to understand?